### PR TITLE
Print useful hdr3 values to screen

### PR DESCRIPTION
**Pull request by jeanconn**
Print useful hdr3 values to screen

### CODE REVIEW

**Comment by taldcroft** [comment]
On a different note I have been playing around a little to add a curvature parameter to the scaling relation to see if this can improve the fit.

On this pr I didn't look too closely but assume this is what you need.

**Comment by jeanconn** [comment]
I think all we really wanted (but didn't have handy during the pathfinder) was the setpoint value as read back out in the hdr3 telemetry.  

I've tested out this code with the telemetry from the pathfinder, using something like:

```
dump2occ.py 2015_252_0210_252_1459_Dump_EM_47053N001.sto | perl splat.pl -noimage
```

And here is some example output during the event:

```
Format: 2  VCDU:   980992 (  7664,000)  

Header: 00 07 00 00 00 9e df bf  int_tim:0007  cal:0 chk:0 syn:0 cnt:00   0 syn:0 cnt:00   
INT_TIME=   0.112 sec  CCD_Temp= -12  AChous_Temp= 26.8                             
Raw 0: 04 00 ab d1 04 7c 13 01 c0 a0 28 0a 02 40 80 1c 07 02 c0 d0 58 16 04 00 b0 28 08 
Raw 1: 14 ff f9 14 bd 1f 85 03 80 e0 4c 34 10 02 60 54 10 03 80 d0 40 15 05 81 10 48 11 
Raw 2: 24 04 e2 43 42 40 ff 04 c2 32 0e d4 d5 5b 10 e8 1c 05 c3 77 73 b2 b4 ff 91 e0 21 
Raw 3: 34 04 e2 43 42 40 ff 07 84 75 82 9c 4c 42 f0 64 13 09 8b 4e fe fd fe c8 10 84 13 
Raw 4: 44 80 1b 7b 25 80 1d 05 01 50 68 36 0f c2 30 5c 15 04 c1 40 5c 1a 06 c1 70 58 14 
Raw 5: 54 eb 24 80 20 9b 45 07 02 e3 b7 f1 91 ff 32 c0 28 06 01 e1 51 1b ba db e1 1c 1d 
Raw 6: 68 20 c7 82 80 fb 62 0c 82 c0 bc 32 0d 03 00 b0 32 0c c3 00 c8 30 0c 83 00 b0 34 
Raw 7: 78 1a 61 00 00 01 d1 0c 43 10 b8 32 0c 03 20 c0 33 0b c2 e0 b4 32 0c 83 60 b8 35 
dac = 465             
ccd_setpoint = -5.00       
ccd_temp = -11.82        

Format: 2  VCDU:   985344 (  7698,000)  

Header: 00 07 00 00 00 9e df bf  int_tim:0007  cal:0 chk:0 syn:0 cnt:00   0 syn:0 cnt:00   
INT_TIME=   0.112 sec  CCD_Temp= -6.8  AChous_Temp= 26.8                            
Raw 0: 04 00 a7 d0 04 70 13 01 c0 90 28 0a 02 80 90 1c 07 02 40 c0 54 13 04 40 d0 28 08 
Raw 1: 14 ff f8 14 ce 1f 90 03 00 d0 7c 40 0a 81 20 2c 0d 02 80 c0 3c 15 03 c0 f0 30 0b 
Raw 2: 24 05 ef 43 42 40 ff 07 c4 f7 57 22 77 84 20 80 11 0a 4e 6f f6 ae f7 8e 20 9c 13 
Raw 3: 34 05 ef 43 42 40 ff 06 03 73 0e 43 4d 03 d0 74 15 07 86 4b d7 53 ff 4f d0 b4 16 
Raw 4: 44 80 1b c6 50 80 1d 05 c1 c0 70 37 16 43 30 6c 1a 05 41 70 58 1c 08 82 00 5c 18 
Raw 5: 54 dc 71 80 20 03 ab 09 47 af 0a 37 ca da f0 c0 1a 06 c3 b3 c2 43 9e 4b 00 80 1a 
Raw 6: 68 20 c7 82 80 fd 55 0c c3 00 d0 30 0c 02 e0 ac 33 0c 82 f0 c0 2e 0d 03 00 c0 35 
Raw 7: 78 1a 4b 00 00 01 6b 0c 03 10 c4 35 0e c2 d0 c4 36 0c 82 e0 c4 2f 0d c3 60 d0 3c 
dac = 363             
ccd_setpoint =  0.00       
ccd_temp = -6.83         




```

I can't think of any reason we'd want the setpoint value in the saved-out per-slot data or the like, so I think this screen output is fine.

I note that it doesn't give perfect results if we aren't stable in 8x8 on the relevant slots.  I also note the ccd temperature value is a bit overkill since we're already getting something good enough from the non-hdr3 temperature value.

**Comment by taldcroft** [comment]
There is the somewhat unresolved question of why the CCD warmed so slowly when we just bumped the setpoint up in 5 C increments.  Do you think having these values in the per-slot data would help us?  I haven't thought about it in detail.

**Comment by jeanconn** [comment]
> There is the somewhat unresolved question of why the CCD warmed so slowly when we just bumped the setpoint up in 5 C increments.

Do we want to try to understand the TEC control law before we do this
again?  I recall you dug up the PEA Forth code.

> Do you think having these values in the per-slot data would help us? I haven't thought about it in detail.

I can't think of any use for them in our real-time fits / plots or the like.

**Comment by taldcroft** [comment]
Looking at the code would probably be useful.  Not a high priority, so this PR looks good to me.

File: splat.pl [hunk 1]
```diff
@@ -804,6 +823,29 @@ sub print_raw {
...
+                     'ccd_temp' => {'val' => $HDR3{6}->{7}->[2],
+                                    'scale' => 0.01,
+                                    'fmt' => "%s = %5.2f        \n"});
+
+    for my $msid ('dac', 'ccd_setpoint', 'ccd_temp'){
+        my $val = $hdr3_want{$msid}->{'val'};
+        my $fmt = $hdr3_want{$msid}->{'fmt'};
+        my $scale = $hdr3_want{$msid}->{'scale'};
+        if (defined $val){
+            $out .= sprintf($fmt, $msid, $scale * unpack("s", pack("S", $val)));
```
Comment (jeanconn) [inline splat.pl:844]: I should probably add a comment here about this unpack/pack magic that is taking the 2-byte value and explicitly converting it to a signed int.

**Pull request merged**